Controllers/ReleaseNoteController.cs

```diff
@@ -25,11 +27,25 @@ public ReleaseNoteController(IReleaseNoteService releaseNoteService, IMapper map
 
         // GET: api/releasenote
         [HttpGet]
-        public async Task<IEnumerable<ReleaseNoteResource>> GetAllAsync()
+        public async Task<ActionResult<ReleaseNoteResource>> GetAllAsync(
+            [FromQuery] ReleaseNoteParameters queryParameters)
         {
-            var releaseNotes = await _releaseNoteService.ListAsync();
+            var releaseNotes = await _releaseNoteService.ListAsync(queryParameters);
+            if (queryParameters.StartDate.HasValue && queryParameters.EndDate.HasValue)
+            {
+                if (queryParameters.StartDate.Value.CompareTo(queryParameters.EndDate.Value) > 0)
+                {
+                    return BadRequest(ModelState);
+                }
+
+                var filteredNotes = await _releaseNoteService.FilterDates(releaseNotes, queryParameters);
+                var releaseNote = _mapper.Map<List<ReleaseNote>, IEnumerable<ReleaseNote>>(filteredNotes.List);
+                var res = _mapper.Map<IEnumerable<ReleaseNoteResource>>(releaseNote);
+                return Ok(res);
```

Comment:
La meg prøv å tolk det som ska skje her.

1. Først hentes release notes ut, med queryparameters
2. Så sjekk vi om det finnes queryparameters.
3. Hvis start- er før sluttdato, returner 400.
4. Filtrer de Release Notes vi henta i steg 1.
5. Map en `List<ReleaseNote>` tell en `IEnumerable<ReleaseNote>`.
6. Map en `IEnumerable<ReleaseNoteResource>` tell en `IEnumerable<ReleaseNote>`.

Kun du ha gjort koden litt enklere? Eg skjønn ikke koffer du kjør `ListAsync()` før du ha sjekka om det finnes `queryParameters`, så trur du må gjennomfør steg 2 og 3 først. 

Også lur eg på koffer du kall `ListAsync()` med `queryParameters`, også filterer du de samme ReleaseNotesan der ett av parameteran e `queryParameters`, sku tru at man slapp å gjør det , men eg har ikke lest heile koden enda.

E det nødvendig å map to ganga? Eg ser ikke helt koffer du e nødt tell å gjør det.

Reply:
ListAsync trenger ikke å ha queryParameters. Hang igjen fra et annet forsøk. Skal fjerne den 
Hvis start skjer ETTER slutt, returner 400*
Skal se om det er noe i veien for å endre rekkefølgen

Reply:
Rekkefølgen er endret nå, funksjonen som er tatt i bruk er også endret litt